SQL-Solutions.txt

```diff
@@ -0,0 +1,2743 @@
+| 2001 | Hoffman | London  |    100 | 1001 | 2006 | Clemens | London  |    100 | 1001 |
+| 2003 | Liu     | SanJose |    200 | 1002 | 2004 | Grass   | Berlin  |    300 | 1002 |
+| 2004 | Grass   | Berlin  |    300 | 1002 | 2003 | Liu     | SanJose |    200 | 1002 |
+| 2006 | Clemens | London  |    100 | 1001 | 2001 | Hoffman | London  |    100 | 1001 |
++------+---------+---------+--------+------+------+---------+---------+--------+------+
+
+-- 34) Select all orders that are greater than the average for October 4. 
+
+Query -
+
+SELECT 
+    *
+FROM
+    ORDERS
+WHERE
+    amt > (SELECT 
+            AVG(amt)
+        FROM
+            orders
+        GROUP BY odate
+        HAVING MONTH(odate) = 10 AND DAY(odate) = 4);
+
+Output -
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3002 |  1900.1 | 1990-10-03 | 2007 | 1004 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3008 |    4723 | 1990-10-05 | 2006 | 1001 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3010 | 1309.95 | 1990-10-06 | 2004 | 1002 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
++------+---------+------------+------+------+
+
+-- 35) Write a select command using correlated subquery that selects the names and 
+-- numbers of all customers with ratings equal to the maximum for their city. 
+
+Query -
+
+SELECT 
+    c1.cnum, c1.cname
+FROM
+    customers AS c1
+WHERE
+    rating = (SELECT 
+            MAX(rating)
+        FROM
+            customers AS c2
+        WHERE
+            c1.city = c2.city);
+
+Output - 
+
++------+----------+
+| cnum | cname    |
++------+----------+
+| 2001 | Hoffman  |
+| 2002 | Giovanni |
+| 2004 | Grass    |
+| 2006 | Clemens  |
+| 2008 | Cisneros |
++------+----------+
+
+-- 36) Write a query that totals the orders for each day and places the results in 
+-- descending order. 
+
+Query -
+
+SELECT 
+    odate, SUM(amt) AS total_order_amount
+FROM
+    orders
+GROUP BY odate
+ORDER BY total_order_amount DESC;
+
+Output -
+
++------------+--------------------+
+| odate      | total_order_amount |
++------------+--------------------+
+| 1990-10-06 |           11201.83 |
+| 1990-10-03 |            8944.59 |
+| 1990-10-05 |               4723 |
+| 1990-10-04 |            1788.98 |
++------------+--------------------+
+
+-- 37) Write a select command that produces the rating followed by the name of each 
+-- customer in SanJose. 
+
+Query -
+
+SELECT 
+    rating, cname, city
+FROM
+    customers
+WHERE
+    city = 'SanJose';
+
+Output - 
+
++--------+----------+---------+
+| rating | cname    | city    |
++--------+----------+---------+
+|    200 | Liu      | SanJose |
+|    300 | Cisneros | SanJose |
++--------+----------+---------+
+
+-- 38) Find all orders with amounts smaller than any amount for a customer in SanJose. 
+
+Query -
+
```

Comment:
Can we do it using Joins? then add that way also.

Reply:
Yes we can do it with Joins. Added that way.